File: Transfer Matrix/Multilayer/testPC.py
import numpy as np
import math
import matplotlib.pyplot as plt
from PC import PC as PC

if __name__ == '__main__':
    N = 20
    c = 3.0*10**8
    lambda0 = c/10**10
    frequency = 10**9*np.arange(0,20,0.01)

    materialid = np.ones(N)
    d = np.ones(N)
    for n in range(N):
        if n%2 == 1:
            materialid[n] = 1
            d[n] = lambda0/4/math.sqrt(12.25)
        else:
            materialid[n] = 2
            d[n] = lambda0/4/math.sqrt(4)


    [reflection,transmission] = PC(materialid,d,frequency)
    plt.plot(1.0*10**-9*frequency,np.square(np.abs(transmission)))
    plt.title('Transmission Spectrum')
    plt.xlabel('Frequency (GHz)')
    plt.ylabel('Transmission')
    plt.ylim(0,1)
    plt.show()

File: Transfer Matrix/julia/testPC.jl
using Plots
include("PC.jl")

N = 20
c = 3e8
lambda0 = c/1e10
frequency = 1e9*(0:0.01:20)
#frequency = 1e9*[1 2]

materialid = ones(N)
d = ones(N)
for n = 1:N
    if n%2 == 0
        materialid[n] = 1
        d[n] = lambda0/4/sqrt(12.25)
    else
        materialid[n] = 2
        d[n] = lambda0/4/sqrt(4)
    end
end

reflection, transmission = PC(materialid, d, frequency)
println("Plotting...")
gr()
display(plot(1e-9*frequency, abs2.(transmission)))